### PR TITLE
fix(order-in-components): allow autofix with TypeScript PropType

### PR DESCRIPTION
**Pull request by acupofspirt**
In the following code `vue/order-in-components` rule can't be autofixed because of `as PropType`:
```ts
import { defineComponent, PropType } from 'vue'

export defineComponent({
  setup () {},

  props: {
    someProp: { 
      type: Array as PropType<string[]>
    }
  }
})
```
TSAsExpression node is considered as not side-effects free. This PR fixes it.

### CODE REVIEW

**Comment by ota-meshi** [comment]
Thank you for this PR. Could you please add test cases?

**Comment by acupofspirt** [comment]
[ota-meshi](https://github.com/ota-meshi) tests done

File: tests/lib/autofix.js [hunk 1]
```diff
@@ -172,3 +172,46 @@ describe('Complex autofix test cases', () => {
     })
   })
 })
+
+describe('Autofix test cases with Typescript', () => {
```
Comment (ota-meshi) [inline tests/lib/autofix.js:176]: Could you change it to test case using rule tester?
https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/rules/order-in-components.js

Comment (acupofspirt) [reply tests/lib/autofix.js]: done!

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Comment by ota-meshi** [comment]
ts-eslint-v4 error is fixed in #1956.

**Review by FloEdelmann — APPROVED** [review]

**Pull request merged**